### DISCUSSION TITLE
how to scale alone for anyone object inside group?

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8633] [General]
i donot want the object inside group is scaled when the group is scaling. for example. i want the scaleX is 0.5 of object  inside group when the group is scaling.i want the scaleX  value is any number of the object inside group.instead of is changed by group.

**Comment by ShaMan123** [comment]
I would recommend using resizing controls instead then, same as textbox does.
#8426

**Reply by wuxingping** [reply]
thanks a lot .but the one inside group still changed(scale) if the group is scaling , am i right?

**Reply by ShaMan123** [reply]
Yes, that is why I suggest using resize logic instead of scale logic.
You can reverse scaling on the children OR apply an inverse transform when rendering the children in `drawObject`

**Reply by ShaMan123** [reply]
> apply an inverse transform when rendering the children in drawObject

But that is not enough since the object will think it is scaled but will be drawn not scaled so you will encounter bugs unless you override more methods (`calcTransformMatrix`)

**Reply by wuxingping** [reply]
> > apply an inverse transform when rendering the children in drawObject
> 
> But that is not enough since the object will think it is scaled but will be drawn not scaled so you will encounter bugs unless you override more methods (`calcTransformMatrix`)

i am so glad, you've been help me a lot ,thank you so much.but how to override calcTransformMatrix?do you have more advice...

**Reply by ShaMan123** [reply]
```
const superMethod = MyObjectClass.prototype.calcTransformMatrix;
MyObjectClass.prototype.calcTransformMatrix = function () {
   return superMethod(true);
}
```

Or sublass

**Reply by wuxingping** [reply]
> ```
> const superMethod = MyObjectClass.prototype.calcTransformMatrix;
> MyObjectClass.prototype.calcTransformMatrix = function () {
>    return superMethod(true);
> }
> ```
> 
> Or sublass

if do like that,does it imfact dragging and rotating  of the object inside group? i mean the object would be not dragged or rotated when the group has dragging or rotating?i want the object to drag or rotate but not scale  when the group has modified.

**Reply by ShaMan123** [reply]
It will, play with it. Maybe invert scaling alone...

**Comment by asturur** [comment]
@wuxingping it would be useful to understand the feature you are working on.
Asking for a particular value to do not change under certain conditions hides the final goal you are going for.
There may be easier answers for the bigger picture ( or maybe not ).
